src/ios/CDVInAppBrowserOptions.m

```diff
@@ -73,7 +74,19 @@ + (CDVInAppBrowserOptions*)parseOptions:(NSString*)options
 
             // set the property according to the key name
             if ([obj respondsToSelector:NSSelectorFromString(key)]) {
-                if (isNumber) {
+                if ([key isEqualToString:@"basicauth"]) {
+                    if ([value isEqualToString:@""]) {
+                        continue;
+                    }
+                    NSString *escapedString = [value stringByReplacingOccurrencesOfString:@"+" withString:@" "];
+                    escapedString = [escapedString stringByRemovingPercentEncoding];
+                    NSData *jsonData = [escapedString dataUsingEncoding:NSUTF8StringEncoding];
+                    NSError *error;
+                    NSDictionary *jsonDictionary = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:&error];
```

Comment:
Awesome, nice url decode and json parse.  Interesting that they have a `stringByRemovingPercentEncoding` method but not a fully `stringByRemovingUrlEncoding` method.

Reply:
TBH, I am still fuzzy on how the decode/encode URL works. Wounded up with this by trail and errror (and some deprecation warnings from Xcode)

Reply:
oh?  which function is deprecated?